nilearn/image/tests/test_image.py

```diff
@@ -514,3 +514,10 @@ def test_clean_img():
 
     np.testing.assert_almost_equal(data_img_.get_data().T.reshape(100, -1),
                                    data_flat_)
+    # if NANs
+    data[:, 9, 9] = np.nan
+    # if infinity
+    data[:, 5, 5] = np.inf
+    nan_img = nibabel.Nifti1Image(data, np.eye(4))
+    clean_im = image.clean_img(nan_img, ensure_finite=True)
```

Comment:
I think image will still have non-finite values and might break somewhere in the analysis.

With the same test code it behaves like this:
```python
nilearn/signal.py:146: RuntimeWarning: invalid value encountered in subtract
  signals -= np.mean(signals, axis=0)
nilearn/signal.py:63: RuntimeWarning: invalid value encountered in less
  std[std < np.finfo(np.float).eps] = 1.  # avoid numerical problems
```